### ISSUE TITLE
isActive with indexOnly=true works unexpectedly

### ISSUE THREAD

**Issue by CoffeeKitty** [rackt/react-router#2238]
With a route configuration like this:

```
<Route component={ComponentA}>
    <Route path='/' component={ComponentB}>
        <IndexRoute component={ComponentC}/>
    </Route>
</Route>
```

`this.context.history.isActive('/', {}, true)` will always return `false`, even when `pathname === "/"`. This seems to be an unexpected behavior to me.

**Comment by knowbody** [comment]
@CoffeeKitty mind sending PR with the failing test?

**Comment by taion** [comment]
I actually hit the same thing and have a fix and tests on #2237.

**Comment by knowbody** [comment]
I actually looked at it @taion and thought that was fixed there. Should I merge that PR or are you waiting for @mjackson?

**Comment by taion** [comment]
Your call. It's not a very big change, I just didn't want to merge my own PR.

**Comment by knowbody** [comment]
thanks @taion, merged. so this should be fixed (#2237) with the next release @CoffeeKitty

**Comment by CoffeeKitty** [comment]
Thanks! :)
